### DISCUSSION TITLE
SDK Batch Transactions

### DISCUSSION THREAD

**Discussion by JohnChangUK** [aptos-labs/aptos-developer-discussions#156] [Questions] [answered]
### Describe your question in detail.

When submitting batch TXs with the SDK, do these get submitted serially (one after the other)? Or parallel, with each TX having a higher sequence number than prev TX. 

**Context**: Calling `entry fun foo()` 5 times, would the timestamp of each TX be unique? (using microseconds). 

Goal is to verify if sending batch transactions will result in the same timestamp generated on chain.

**Comment by 0xmaayan** [comment, marked as answer]
The batch transaction management in the SDK sends transactions in parallel. First it fetches the account sequence number from chain and maintains it in a local queue so then it sends the next transaction with a higher sequence number than the prev transaction.

Can check the code here https://github.com/aptos-labs/aptos-ts-sdk/blob/main/src/transactions/management/transactionWorker.ts#L60
